src/main/java/org/jabref/logic/xmp/DocumentInformationExtractor.java

```diff
@@ -0,0 +1,99 @@
+package org.jabref.logic.xmp;
+
+import java.util.Map;
+import java.util.Optional;
+
+import org.jabref.model.entry.BibEntry;
+import org.jabref.model.entry.FieldName;
+
+import org.apache.pdfbox.cos.COSBase;
+import org.apache.pdfbox.cos.COSDictionary;
+import org.apache.pdfbox.cos.COSName;
+import org.apache.pdfbox.pdmodel.PDDocumentInformation;
+
+public class DocumentInformationExtractor {
+
+    private final PDDocumentInformation documentInformation;
+
+    private final BibEntry bibEntry;
+
+    public DocumentInformationExtractor(PDDocumentInformation documentInformation) {
+        this.documentInformation = documentInformation;
+
+        this.bibEntry = new BibEntry();
+    }
+
+    private void extractAuthor() {
+        String s = documentInformation.getAuthor();
+        if (s != null) {
+            bibEntry.setField(FieldName.AUTHOR, s);
+        }
+    }
+
+    private void extractTitle() {
+        String s = documentInformation.getTitle();
+        if (s != null) {
+            bibEntry.setField(FieldName.TITLE, s);
+        }
+    }
+
+    private void extractKeywords() {
+        String s = documentInformation.getKeywords();
+        if (s != null) {
+            bibEntry.setField(FieldName.KEYWORDS, s);
+        }
+    }
+
+    private void extractSubject() {
+        String s = documentInformation.getSubject();
+        if (s != null) {
```

Comment:
We once agreed on avoiding single char variables. Please use a meaningful name here and for the others

Reply:
Done.